bots/tests-scan

```diff
@@ -139,22 +145,20 @@ def main():
     api = github.GitHub(repo=opts.repo)
 
     try:
+        policy = REPO_BRANCH_CONTEXT[opts.repo]
         if opts.context:
-            policy = { c: [ "master" ] for c in opts.context }
-        elif opts.repo:
             policy = {}
-            if opts.repo in EXTERNAL_PROJECTS:
-                policy.update({ c: [ "master" ] for c in EXTERNAL_PROJECTS[opts.repo] })
-            if opts.repo in REDHAT_EXTERNAL_PROJECTS:
-                policy.update({ c: [ "master" ] for c in REDHAT_EXTERNAL_PROJECTS[opts.repo] })
-            if not policy:
-                policy = default_policy()
-        else:
-            policy = default_policy()
+            for branch in REPO_BRANCH_CONTEXT[opts.repo]:
+                branch_context = []
+                for context in opts.context:
+                    if context in REPO_BRANCH_CONTEXT[opts.repo][branch]:
+                        branch_context.append(context)
+                if branch_context:
+                    policy[branch] = branch_context
         results = scan_for_pull_tasks(api, policy, opts, None)
 
-        # for the default (Cockpit) repo, also check external project PRs
-        if not opts.repo:
+        # for the Cockpit repo, also check external project PRs
+        if opts.repo == "cockpit-project/cockpit":
```

Comment:
This is a good change -- it's actively wrong in current master when you run it in a repo != cockpit. This is the legit special case that we still have while bots/ is part of the cockpit repo. (Just use `api.repo`, as discussed above).